### PR TITLE
feat: database and models

### PR DESCRIPTION
**Pull request by SiongSng**
# Description
Connect to the database.
Add user and lesson model
Using `vitest` package for unit testing

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by lumynou5** [comment]
The readme needs more information about configuring.

**Review by lumynou5 — COMMENTED** [review]
Improve the `.env.example` please.
Furthermore, the major version number of vitest is 0; that is, it's easy to make breaking changes. Therefore, the version should be locked in a minor version instead of a major version.

**Comment by SiongSng** [comment]
> Improve the `.env.example` please. Furthermore, the major version number of vitest is 0; that is, it's easy to make breaking changes. Therefore, the version should be locked in a minor version instead of a major version.

done

File: src/index.ts [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
-const app = createServer();
-const port = process.env.PORT ?? 8080;
+export let db: Database;
 
-app.listen(port, () => {
-  console.log(`Server is running at https://localhost:${port}`);
-});
+async function main() {
+  const app = createServer();
+  const port = process.env.PORT ?? 8080;
```
Comment (banahaker) [inline src/index.ts]: i think is `||`

**Review by banahaker — CHANGES_REQUESTED** [review]
a liiiiiiiitle problem

File: src/index.ts (hunk 1 above)
Comment (SiongSng) [reply src/index.ts]: sorry
I made a stupid problem

Comment (lumynou5) [reply src/index.ts]: > i think is `||`

I'm not sure wdym. It's an operator in TypeScript that returns the default value (rhs) if the lhs expression is null or undefined. `||` will return the rhs if the lhs is a falsy value, including `false`, `0`, empty string, etc.

**Review by banahaker — APPROVED** [review]

**Pull request merged**